controllers/reaper/reaper_controller.go

```diff
@@ -248,6 +298,29 @@ func (r *ReaperReconciler) reconcileDeployment(
 	return ctrl.Result{}, nil
 }
 
+func getDeploymentReplicas(actualDeployment client.Object, logger logr.Logger) int32 {
+	switch actual := actualDeployment.(type) {
+	case *appsv1.Deployment:
+		return *actual.Spec.Replicas
+	case *appsv1.StatefulSet:
+		return *actual.Spec.Replicas
+	default:
+		logger.Error(fmt.Errorf("unexpected type %T", actualDeployment), "Failed to get deployment replicas")
```

Comment:
**Suggestion:** we should probably fail the reconcile here instead of swallowing the error. The return type should then be `(int32, error)` and if an error occurs it should be returned.

Reply:
Pushed a commit solving this.